File: problems/463/solution.go
package main
func islandPerimeter(grid [][]int) int {
    if len(grid) == 0 || len(grid[0]) == 0 {
        return 0
    }
    
    p := 0
    
    for i := 0; i < len(grid); i++ {
        for j := 0; j < len(grid[0]); j++ {
            if grid[i][j] == 0 {
                continue
            }
            
            if i == 0 || grid[i - 1][j] == 0 {
                p++
            }
            
            if j == 0 || grid[i][j - 1] == 0 {
                p++
            }
            
            if i == len(grid) - 1 || grid[i + 1][j] == 0 {
                p++
            }
            
            if j == len(grid[0]) - 1 || grid[i][j + 1] == 0 {
                p++
            }
        }
    }
    
    return p
}

File: problems/463/solution.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int ans = 0, m = size(grid), n = size(grid.back());
        
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (not grid[i][j]) {
                    continue;
                }
                
                ans += not i or not grid[i - 1][j];
                ans += not j or not grid[i][j - 1];
                ans += i == m - 1 or not grid[i + 1][j];
                ans += j == n - 1 or not grid[i][j + 1];
            }
        }
        
        return ans;
    }
};